sphinx/builders/linkcheck.py

```diff
@@ -441,21 +438,39 @@ def write_linkstat(self, data: dict) -> None:
         self.json_outfile.write(json.dumps(data))
         self.json_outfile.write('\n')
 
+    def queue_hyperlinks(self) -> int:
```

Comment:
I think it would be better to collect anchors in HyperlinkCollector instead.

Reply:
Thanks for the review!

The `linkcheck_ignore` should be checked before the anchors are split (see https://github.com/sphinx-doc/sphinx/pull/8830/commits/a7f31f79ac95bc570463ec129bd4d662ec6892f8).
Performing the ignore checks in the `HyperlinkCollector` brings use closer to https://github.com/sphinx-doc/sphinx/pull/8774, which has the downside of needing to report results from the collector. Perhaps we could use a separate container for ignored URI, so that the builder handles the logging iterates on that container and logs the results?

Reply:
>The linkcheck_ignore should be checked before the anchors are split (see a7f31f7).

Ah, okay. I understand. I did not notice that.